resources/assets/components/CampaignSingle/index.js

```diff
@@ -145,24 +146,29 @@ class CampaignSingle extends React.Component {
         {this.props.loading || this.state.loadingNewPosts ?
           <div className="spinner"></div>
         :
-
-          map(this.props.postIds, (key, value) => {
-            var post = find(posts, {'id': key});
-
-            return <Post key={key}
-              post={post}
-              user={signups[post.signup_id].user.data}
-              signup={signups[post.signup_id]}
-              campaign={campaign}
-              onUpdate={this.props.updatePost}
-              onTag={this.props.updateTag}
-              deletePost={this.props.deletePost}
-              showHistory={this.props.showHistory}
-              rotate={this.props.rotate}
-              showSiblings={true}
-              showQuantity={true}
-              allowHistory={true} />;
-          })
+          this.props.postIds.length !== 0 ?
+            map(this.props.postIds, (key, value) => {
+              var post = find(posts, {'id': key});
+
+              return <Post key={key}
+                post={post}
+                user={signups[post.signup_id].user.data}
+                signup={signups[post.signup_id]}
+                campaign={campaign}
+                onUpdate={this.props.updatePost}
+                onTag={this.props.updateTag}
+                deletePost={this.props.deletePost}
+                showHistory={this.props.showHistory}
+                rotate={this.props.rotate}
+                showSiblings={true}
+                showQuantity={true}
+                allowHistory={true} />;
+            })
+          :
+            <Empty details={{
+              "header":  "There are no results!",
```

Comment:
It'd be nice to have these each as individual `header` and `copy` props so we can use types to validate their input in the future or set default values if nothing is provided. In general, I'm somewhat wary of sending object props unless giving an API response to a "data" component.